File: src/id0064.c
// Licensed under the MIT License.

// Odd Period Square Roots

#include "../lib/euler.h"

int main(void)
{
    int count = 0;
    clock_t start = clock();

    for (int s = 2; s < 10000; s++)
    {
        long isqrt;

        if (math_is_polygonal(4, s, &isqrt))
        {
            continue;
        }

        int a = isqrt;
        int h = 0;
        int i = 0;
        int k = 1;
        int end = 2 * a;

        while (a != end)
        {
            h = a * k - h;
            k = (s - h * h) / k;
            a = (isqrt + h) / k;
            i++;
        }

        count += i % 2;
    }

    return euler_submit(64, count, start);
}

File: src/id0064.py
# Licensed under the MIT License.

# Odd Period Square Roots

from sympy import integer_nthroot
from time import time

start = time()
count = 0

for n in range(2, 10000):
    isqrt, is_square = integer_nthroot(n, 2)
    
    if is_square: continue

    h = 0
    k = 1
    i = 0
    a = isqrt
    end = 2 * a
    
    while a != end:
        h = a * k - h
        k = (n - h * h) // k
        a = (isqrt + h) // k
        i += 1
    
    count += i % 2

print(f"0064{count:>64}    {time() - start:.6f}")
